From 00f3fcb7297c444e694a58577845cc876b07ac9f Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Wed, 31 Oct 2012 18:41:46 +0900
Subject: [PATCH 90/97] dfb-video-example: remove duplicate test to determine
 if audio decoder was sucessfully created.

The test doesn't need to be duplicated in each branch of the 'if' statement.
Once, after the 'if' is cleaner.
---
 ext/directfb/dfb-video-example.c |   13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/ext/directfb/dfb-video-example.c b/ext/directfb/dfb-video-example.c
index c2836ec..07b4090 100644
--- a/ext/directfb/dfb-video-example.c
+++ b/ext/directfb/dfb-video-example.c
@@ -159,17 +159,9 @@ create_audio_pipeline (GstPad * pad)
     if (ver == 4 || ver == 2) { /* AAC */
       printf ("codec type video/mpeg mpegversion=%d\n", ver);
       decoder = gst_element_factory_make ("faad", NULL);
-      if (!decoder) {
-        printf ("faad plugin wasn't found\n");
-        return;
-      }
     } else if (ver == 1 && layer == 3) {        /* MP3 */
       printf ("codec type video/mpeg mpegversion=1\n");
       decoder = gst_element_factory_make ("mad", NULL);
-      if (!decoder) {
-        printf ("mad plugin wasn't found\n");
-        return;
-      }
     } else {
       printf ("unsupported format\n");
       return;
@@ -179,6 +171,11 @@ create_audio_pipeline (GstPad * pad)
     return;
   }
 
+  if (!decoder) {
+    printf ("audio decoder plugin wasn't found\n");
+    return;
+  }
+
   queue = gst_element_factory_make ("queue", NULL);
   g_assert (queue);
   sink = gst_element_factory_make ("alsasink", NULL);
-- 
1.7.10.4

